### DISCUSSION TITLE
Rocks with Lazy.nvim with Rockspec disabled removed the Rocks command.

### DISCUSSION THREAD

**Discussion by andrewcrook** [lumen-oss/rocks.nvim#496] [Q&A] [answered]
If I load lazy and then rocks.nvim
Then configure Lazy also disabling Rockspec Lazy support under lazy.setup()
  
``` 
sources = {
      "lazy",
      --"rockspec", -- disable  experimental  Lazy LuaRocks we use rocks.nvim
      --"packspec", -- not used yet experimental
    },
  },
```
The ‘:Rocks' command is removed for rocks.nvim.

I then have to call the following to get it back

```
 require(“rocks.commands").create_commands()
```

I wonder if there is a cleaner way to do this?

**Comment by mrcjkb** [comment, marked as answer]
Hey :wave: 

lazy.nvim disables Neovim's built-in plugin loading mechanism, which prevents rocks.nvim from being loaded.
If you're not using lazy.nvim to install rocks.nvim (which I guess isn't possible with its luarocks support disabled), it won't load it.

You could try the following lazy.nvim spec:

```lua
{
  "nvim-neorocks/rocks.nvim",
  dir = "<path-to-rocks-nvim-luaorkcs-installation>", -- e.g. "/home/<user>/.local/share/nvim/rocks/lib/luarocks/rocks-5.1/rocks.nvim/2.36.1-1""
  lazy = false,
}
```

The problem is that luarocks installs the plugin's initialisation scripts into a `<version>` subdirectory
(`rocks.nvim/2.36.1-1` in this example), so you have to determine the `dir` programmatically.

**Reply by andrewcrook** [reply]
yeah I enabled luarocks in lazy and rocks.nvim wouldn’t install properly loads of errors it also downloaded toml support for the wrong  platform. I will have to go for one or the other as both work on their own. It’s a shame,  I think I am going to have to try the rocks support in lazy at the moment sorry.

**Reply by mrcjkb** [reply]
> it also downloaded toml support for the wrong platform

To work around that, you can [set `arch` in your luarocks config](https://github.com/luarocks/luarocks/wiki/Config-file-format#platform-specific-settings).

> I think I am going to have to try the rocks support in lazy at the moment

I don't know what other issues you might run into when using both together, so that's probably a good idea.

**Reply by andrewcrook** [reply]
Still having trouble I decided to get rid of lazy and try rocks.nvim as the package manager
